pkg/operator/controller/dns_status.go

```diff
@@ -70,14 +70,14 @@ func computeDNSDegradedCondition(oldCondition *operatorv1.OperatorCondition, clu
 		degradedCondition.Status = operatorv1.ConditionTrue
 		degradedCondition.Reason = "NoClusterIP"
 		degradedCondition.Message = "No ClusterIP assigned to DNS Service"
-	case ds.Status.NumberAvailable == 0:
+	case ds.Status.DesiredNumberScheduled == 0:
```

Comment:
This should stay `NumberAvailable == 0`, right?

Reply:
i.e. wanting more than one pod is not enough, you actually have to *have* more than one pod to get out of `NoPodsScheduled`

Reply:
NoPodsScheduled is actually the misnomer, the check is intentional... if none are even _desired_ to be scheduled, then available seems irrelevant. I'm not even sure how we can get here in the wild...

Reply:
NoPodsDesired?

Reply:
Maybe DesiredNumberScheduled=0 if none of the nodes match the scheduling criteria due to taints or something.